pkg/cache/queue/cluster_queue.go

```diff
@@ -339,6 +349,26 @@ func (c *ClusterQueue) Pop() *workload.Info {
 	return c.inflight
 }
 
+// rebuildAll rebuilds the entire heap. Must be called with lock held.
+func (c *ClusterQueue) rebuildAll() {
+	for _, wl := range c.heap.List() {
+		c.heap.PushOrUpdate(wl)
+	}
+}
+
+func (c *ClusterQueue) hasPendingPenalties() bool {
+	if c.afsEntryPenalties == nil {
+		return false
+	}
+
+	for lqKey := range c.localQueuesInClusterQueue {
+		if _, hasPenalty := c.afsEntryPenalties.Get(lqKey); hasPenalty {
+			return true
+		}
```

Comment:
If think we should check if the value in map is != 0, because what may happen is that we add a LQ to the map, then subtract the usage, but the key would still exist